pkg/repo/repo.go

```diff
@@ -31,13 +32,26 @@ import (
 	"knative.dev/pkg/logging"
 )
 
-// CreateRepo creates and initializes a Tuf repo for Sigstore by adding
-// keys to bytes. keys are typically for a basic setup like:
-// "fulcio_v1.crt.pem" - Fulcio root cert in PEM format
-// "ctfe.pub" - CTLog public key in PEM format
-// "rekor.pub" - Rekor public key in PEM format
-// but additional keys can be added here.
-func CreateRepo(ctx context.Context, files map[string][]byte) (tuf.LocalStore, string, error) {
+// TargetWithMetadata describes a TUF target with the given Name, Bytes, and
+// CustomMetadata
+type TargetWithMetadata struct {
+	Name           string
+	Bytes          []byte
+	CustomMetadata []byte
+}
+
+type CustomMetadata struct {
+	Usage  string `json:"usage"`
+	Status string `json:"status"`
+}
```

Comment:
Since creating that script, `uri` was also added as custom metadata (for delegations I believe). See https://github.com/sigstore/root-signing/blob/7187cb661647a8420c1d2ebb354bc2b7b8cade37/ceremony/2022-10-18/repository/targets.json#L28 for an example

Can we add that here?

Reply:
yep, done